google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/reframing/ReframingResponseObserver.java

```diff
@@ -158,11 +154,18 @@ private void onRequest(int count) {
     Preconditions.checkState(!autoFlowControl, "Auto flow control enabled");
     Preconditions.checkArgument(count > 0, "Count must be > 0");
 
-    synchronized (lock) {
-      int maxCount = Integer.MAX_VALUE - numPending;
-      count = Math.min(maxCount, count);
+    while (true) {
+      int current = numRequested.get();
+      if (current == Integer.MAX_VALUE) {
+        return;
+      }
+      // Prevent overflow
```

Comment:
Consider using this? http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/math/IntMath.html#saturatedAdd-int-int-

Reply:
done